src/helper/style-path.js

```diff
@@ -430,42 +477,62 @@ const getColorsFromSelection = function (selectedItems, bitmapMode) {
             if (itemFillColor2String !== selectionFillColor2String) {
                 selectionFillColor2String = MIXED;
             }
-            if (itemGradientType !== selectionGradientType) {
-                selectionGradientType = GradientTypes.SOLID;
+            if (itemFillGradientType !== selectionFillGradientType) {
+                selectionFillGradientType = GradientTypes.SOLID;
                 selectionFillColorString = MIXED;
                 selectionFillColor2String = MIXED;
             }
+            if (itemStrokeGradientType !== selectionStrokeGradientType) {
+                selectionStrokeGradientType = GradientTypes.SOLID;
+                selectionStrokeColorString = MIXED;
+                selectionStrokeColor2String = MIXED;
+            }
             if (itemStrokeColorString !== selectionStrokeColorString) {
                 selectionStrokeColorString = MIXED;
             }
+            if (itemStrokeColor2String !== selectionStrokeColor2String) {
+                selectionStrokeColor2String = MIXED;
+            }
             const itemStrokeWidth = itemStrokeColorString ? item.strokeWidth : 0;
             if (selectionStrokeWidth !== itemStrokeWidth) {
                 selectionStrokeWidth = null;
             }
         }
     }
     // Convert selection gradient type from horizontal to vertical if first item is exactly vertical
```

Comment:
"This is because we assume all horizontal or vertical gradients are horizontal when comparing gradient types."